Userland/Libraries/LibJS/Tests/builtins/Temporal/PlainTime/PlainTime.prototype.round.js

```diff
@@ -0,0 +1,113 @@
+
+        const secondRoundedPlainTime = plainTime.round({
+            smallestUnit: "minute",
+            roundingMode: "ceil",
+        });
+        expect(secondRoundedPlainTime.hour).toBe(18);
+        expect(secondRoundedPlainTime.minute).toBe(16);
+        expect(secondRoundedPlainTime.second).toBe(0);
+        expect(secondRoundedPlainTime.millisecond).toBe(0);
+        expect(secondRoundedPlainTime.microsecond).toBe(0);
+        expect(secondRoundedPlainTime.nanosecond).toBe(0);
+
+        const thirdRoundedPlainTime = plainTime.round({
+            smallestUnit: "minute",
+            roundingMode: "ceil",
+            roundingIncrement: 30,
+        });
+        expect(thirdRoundedPlainTime.hour).toBe(18);
+        expect(thirdRoundedPlainTime.minute).toBe(30);
+        expect(thirdRoundedPlainTime.second).toBe(0);
+        expect(thirdRoundedPlainTime.millisecond).toBe(0);
+        expect(thirdRoundedPlainTime.microsecond).toBe(0);
+        expect(thirdRoundedPlainTime.nanosecond).toBe(0);
+
+        const fourthRoundedPlainTime = plainTime.round({
+            smallestUnit: "minute",
+            roundingMode: "floor",
+            roundingIncrement: 30,
+        });
+        expect(fourthRoundedPlainTime.hour).toBe(18);
+        expect(fourthRoundedPlainTime.minute).toBe(0);
+        expect(fourthRoundedPlainTime.second).toBe(0);
+        expect(fourthRoundedPlainTime.millisecond).toBe(0);
+        expect(fourthRoundedPlainTime.microsecond).toBe(0);
+        expect(fourthRoundedPlainTime.nanosecond).toBe(0);
+
+        const fifthRoundedPlainTime = plainTime.round({
+            smallestUnit: "hour",
+            roundingMode: "halfExpand",
+            roundingIncrement: 4,
+        });
+        expect(fifthRoundedPlainTime.hour).toBe(20);
+        expect(fifthRoundedPlainTime.minute).toBe(0);
+        expect(fifthRoundedPlainTime.second).toBe(0);
+        expect(fifthRoundedPlainTime.millisecond).toBe(0);
+        expect(fifthRoundedPlainTime.microsecond).toBe(0);
+        expect(fifthRoundedPlainTime.nanosecond).toBe(0);
+    });
+});
+
+describe("errors", () => {
+    test("this value must be a Temporal.PlainTime object", () => {
+        expect(() => {
+            Temporal.PlainTime.prototype.round.call("foo", {});
+        }).toThrowWithMessage(TypeError, "Not an object of type Temporal.PlainTime");
+    });
+
+    test("missing options object", () => {
+        expect(() => {
+            const plainTime = new Temporal.PlainTime(18, 15, 9, 100, 200, 300);
+            plainTime.round();
+        }).toThrowWithMessage(TypeError, "Required options object is missing or undefined");
+    });
+
+    test("invalid rounding mode", () => {
+        expect(() => {
+            const plainTime = new Temporal.PlainTime(18, 15, 9, 100, 200, 300);
+            plainTime.round({ smallestUnit: "second", roundingMode: "serenityOS" });
+        }).toThrowWithMessage(RangeError, "is not a valid value for option roundingMode");
```

Comment:
Please don't rely on "message string but it's actually a regex" for new tests. Just provide the full expected error message.